<?xml version="1.0"?> 

<robot name="link_example">

  <link name="map" />        
  <link name="odom" />  
  <link name="base_footprint" />
  <link name="base_link" />
  <link name="wheel_r_link" />
  <link name="wheel_l_link" />
  <link name="caster_link" />

  <joint name="map_to_odom" type="fixed">
      <parent link="map"/>
      <child link="odom"/>
      <origin xyz="-0.15 0 0" rpy="0 0 0"/>
  </joint>

  <joint name="odom_to_basefootprint" type="fixed">
      <parent link="odom"/>         
      <child link="base_footprint"/>           
      <origin xyz="-0.12 0 0" rpy="0 0 0"/>      
  </joint>

  <joint name="base_link_joint" type="fixed">
      <parent link="base_footprint"/>
      <child link="base_link"/>
      <origin xyz="0 0 0.05" rpy="0 0 0" />
  </joint>

  <joint name="wheel_r_joint" type="continuous">
      <parent link="base_link"/>
      <child link="wheel_r_link"/>
      <origin xyz="0.052 -0.095 -0.0025" rpy="0 0 0" />
      <axis xyz= '1 0 0' />
  </joint>

  <joint name="wheel_l_joint" type="continuous">
      <parent link="base_link"/>
      <child link="wheel_l_link"/>
      <origin xyz="0.052 0.095 -0.0025" rpy="0 0 0" />
      <axis xyz= '1 0 0' />
  </joint>

  <joint name="caster_joint" type="fixed">
      <parent link="base_link"/>
      <child link="caster_link"/>
      <origin xyz="-0.095 0.0 -0.03" rpy="0 0 0" />
  </joint>
    
</robot>  
